packages/paste-core/components/skeleton-loader/src/index.tsx

```diff
@@ -0,0 +1,114 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {useSpring, animated} from '@twilio-paste/animation-library';
+import {css, styled} from '@twilio-paste/styling-library';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import type {LayoutProps, MarginProps, BorderRadiusProps} from '@twilio-paste/style-props';
+
+const AnimatedSkeleton = animated(Box);
+
+const StyledAnimatedSkeleton = styled(AnimatedSkeleton)(() =>
+  css({
+    background: `linear-gradient(90deg, transparent, rgba(255, 255, 255, 0.3) 40%, rgba(255, 255, 255, 0.3) 60%, transparent)`,
+  })
+);
+
+export interface SkeletonLoaderProps
+  extends React.HTMLAttributes<HTMLDivElement>,
+    LayoutProps,
+    MarginProps,
+    BorderRadiusProps {
+  disabled?: boolean;
+}
+
+const SkeletonLoader = React.forwardRef<HTMLDivElement, SkeletonLoaderProps>(
+  (
+    {
+      borderBottomLeftRadius,
+      borderBottomRightRadius,
+      borderRadius = 'borderRadius20',
+      borderTopLeftRadius,
+      borderTopRightRadius,
+      display,
+      disabled,
+      height = 'sizeIcon20',
+      margin,
+      marginBottom,
+      marginLeft,
+      marginRight,
+      marginTop,
+      marginX = null,
```

Comment:
Why do we need to explicitly set these to null?

Reply:
If we don't the other margin props get overridden. Same with size, if not null, it will override height and width.